src/util/address.rs

```diff
@@ -405,6 +405,140 @@ impl Payload {
             } => script::Script::new_witness_program(version, prog)
         }
     }
+
+    /// Creates a pay to (compressed) public key hash payload from a public key
+    #[inline]
+    pub fn p2pkh(pk: &ecdsa::PublicKey) -> Payload {
+        Payload::PubkeyHash(PubkeyHash::hash(&pk.to_bytes()))
+    }
+
+    /// Creates a pay to script hash P2SH payload from a script
+    #[inline]
+    pub fn p2sh(script: &script::Script) -> Result<Payload, Error> {
+        if script.len() > MAX_SCRIPT_ELEMENT_SIZE {
+            return Err(Error::ExcessiveScriptSize);
+        }
+        Ok(Payload::ScriptHash(ScriptHash::hash(&script[..])))
+    }
+
+    /// Create a witness pay to public key payload from a public key
+    pub fn p2wpkh(pk: &ecdsa::PublicKey) -> Result<Payload, Error> {
+        if !pk.compressed {
+            return Err(Error::UncompressedPubkey);
+        }
+
+        Ok(Payload::WitnessProgram {
+            version: WitnessVersion::V0,
+            program: WPubkeyHash::hash(&pk.to_bytes())[..].to_vec(),
+        })
```

Comment:
The whole thing should be done in just a single line with `pk.wpubkey_hash().ok_or(Error::UncompressedPubkey)` which already returns option on a non-compressed keys

Reply:
done